src/native-select/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { focus } from '@dojo/framework/core/middleware/focus';
+import { i18n } from '@dojo/framework/core/middleware/i18n';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { uuid } from '@dojo/framework/core/util';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import HelperText from '../helper-text';
+import theme from '../middleware/theme';
+import * as css from '../theme/default/native-select.m.css';
+import Icon from '../icon';
+import Label from '../label';
+
+export type MenuOption = { value: string; label?: string; disabled?: boolean };
+
+export interface NativeSelectProperties {
+	/** Callback called when user selects a value */
+	onValue?(value: string): void;
+	/** The initial selected value */
+	initialValue?: string;
+	/** Options to display within the menu */
+	options: MenuOption[];
+	/** Property to determine if the input is disabled */
+	disabled?: boolean;
+	/** Sets the helper text of the input */
+	helperText?: string;
+	/** The label to show */
+	label?: string;
+	/** Boolean to indicate if field is required */
+	required?: boolean;
+	/** Used to specify the name of the control */
+	name?: string;
+	/** Represents the number of rows the are visible at one time */
+	size?: number;
+	/** Handler for events triggered by select field losing focus */
+	onBlur?(): void;
+	/** Handler for events triggered by the select element being focused */
+	onFocus?(): void;
+}
+
+interface NativeSelectICache {
+	initial: string;
+	selectId: string;
+	value: string;
+}
+
+const icache = createICacheMiddleware<NativeSelectICache>();
+
+const factory = create({ icache, focus, theme, i18n }).properties<NativeSelectProperties>();
+
+export const NativeSelect = factory(function NativeSelect({
+	properties,
+	middleware: { icache, theme }
+}) {
+	const {
+		classes,
+		disabled,
+		helperText,
+		initialValue,
+		label,
+		onValue,
+		options,
+		required,
+		name,
+		size,
+		onFocus,
+		onBlur,
+		theme: themeProp
+	} = properties();
+
+	if (initialValue !== undefined && initialValue !== icache.get('initial')) {
+		icache.set('initial', initialValue);
+		icache.set('value', initialValue);
+	}
+
+	const selectedValue = icache.get('value');
+	const selectId = icache.getOrSet('selectId', uuid());
```

Comment:
rather than generating this using `uuid` you can actually use the `id` that's passed into the widget via the factory. An example of this can be seen here:  https://github.com/dojo/widgets/blob/master/src/menu/index.tsx#L289

Reply:
taken from the factory function here: https://github.com/dojo/widgets/blob/master/src/menu/index.tsx#L89

Reply:
Cool, I'll update this as well 👍

Reply:
Done